core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -437,10 +437,13 @@ object TopicCommand extends Logging {
     * @param topics
     * @param topicOptWithExists
     */
-  private def ensureTopicExists(topics: Seq[String], topicOptWithExists: Boolean = false) = {
-    if (topics.isEmpty && !topicOptWithExists) {
+  private def ensureTopicExists(topics: Seq[String], desiredTopicName: Option[String], topicOptWithExists: Boolean = false) = {
```

Comment:
Might just be me, but I find this method a bit hard to understand. Maybe it is just a matter of choosing better names. Maybe something like this:

`topics` -> `foundTopics`
`desiredTopicName` -> `requestedTopic`
`topicOptWithExists` -> `requireTopicExists`

Then the check becomes:

```scala
requestedTopic.isDefined && requireTopicExists && foundTopics.isEmpty
```
That seems a little easier to understand. Maybe we can add a simple description for each parameter in the doc above.

Reply:
That seems great. I will make these changes.

Reply:
Changed.